src/mapgen_valleys.cpp

```diff
@@ -25,24 +25,20 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
-
-#include "mapgen.h"
-#include "voxel.h"
-#include "noise.h"
-#include "mapblock.h"
-#include "mapnode.h"
-#include "map.h"
 #include "content_sao.h"
-#include "nodedef.h"
-#include "voxelalgorithms.h"
-#include "settings.h" // For g_settings
-#include "emerge.h"
 #include "dungeongen.h"
-#include "treegen.h"
+#include "emerge.h"
+#include "mapblock.h"
+#include "mapgen_valleys.h"
 #include "mg_biome.h"
-#include "mg_ore.h"
 #include "mg_decoration.h"
-#include "mapgen_valleys.h"
+#include "mg_ore.h"
+#include "treegen.h"
+#include "voxel.h"
+#include "voxelalgorithms.h"
+// cavegen.h must be included after mapgen.h,
```

Comment:
What?? Why is this?

You should never have an order dependency for header files if you can help it.